### PR TITLE
updating models for required fields

### PR DESCRIPTION
**Pull request by ANISH-GOTTAPU**
updating models for required fields and removing defaults

### CODE REVIEW

**Review by ajbalogh — COMMENTED** [review]
Is there a specific issue that is the reason for this change?
Just looking for justification for why defaults are being removed and replaced with required.

**Comment by ANISH-GOTTAPU** [comment]
@ajbalogh : with this PR https://github.com/open-traffic-generator/snappi/pull/72 , if user is not passing the value for properties, defaults are set.

- container_name: port 1
  ethernet:
    ipv4:
      address: 0.0.0.0
      gateway: 0.0.0.0
      name: ipv4
      prefix: 24
    mac: 00:00:fa:ce:fa:ce
    mtu: 1200
    name: eth
  name: device
ports:
- location: 10.39.65.230;4;1
  name: port 1

Earlier, the values were null and ixNetwork used to put ixNetwork defaults, but with new Snappi Version the values are set to `0.0.0.0` which doesn't make much sense. So, we had internal discussion and thought to make them as required and catch them as errors on the client instead of adding not desirable values on the server

File: device/bgp/bgpv4.yaml [hunk 1]
```diff
@@ -7,16 +7,15 @@ components:
       description: >-
         Container for emulated BGPv4 peers and routes.
       type: object
+      required: [dut_address]
       properties:
         local_address:
```
Comment (ajbalogh) [inline device/bgp/bgpv4.yaml:12]: removal of default should make this a required property

File: device/bgp/bgpv6.yaml [hunk 2]
```diff
@@ -7,18 +7,17 @@ components:
       description: >-
         Container for BGPv6 peer settings and routes.
       type: object
+      required: [dut_address]
```
Comment (ajbalogh) [inline device/bgp/bgpv6.yaml:10]: removal of default should make this a required property

**Review by ajbalogh — CHANGES_REQUESTED** [review]
There are properties that have had the defaults removed, shouldn't these properties should now be required.

File: device/bgp/bgpv4.yaml (hunk 1 above)
Comment (ANISH-GOTTAPU) [reply device/bgp/bgpv4.yaml:12]: Most of tgens would take local address from ipv4 stack and populate the local address. removal of default would make it to None and the tgen default values are applied.

File: device/bgp/bgpv6.yaml (hunk 2 above)
Comment (ANISH-GOTTAPU) [reply device/bgp/bgpv6.yaml:10]: Most of tgens would take local address from ipv4 stack and populate the local address. removal of default would make it to None and the tgen default values are applied.

Comment (ajbalogh) [reply device/bgp/bgpv6.yaml:10]: @ANISH-GOTTAPU the comment that concrete implementations would take the local address from the ipv4 stack and apply it to the local address is implicit, the model still requires that property to either have a default value or be required.

File: device/bgp/bgpv4.yaml (hunk 1 above)
Comment (ashutshkumr) [reply device/bgp/bgpv4.yaml:12]: `local_address` is now a required field

File: device/bgp/bgpv6.yaml (hunk 2 above)
Comment (ashutshkumr) [reply device/bgp/bgpv6.yaml]: `local_address` is now a required field

**Review by ashutshkumr — APPROVED** [review]
Add some fields to required as discussed.

**Pull request merged**